core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpDocServiceTest.java

```diff
@@ -117,35 +127,50 @@ private static void addFooMethodInfo(Map<Class<?>, Set<MethodInfo>> methodInfos)
         final EndpointInfo endpoint = new EndpointInfoBuilder("*", "exact:/service/foo")
                 .availableMimeTypes(MediaType.JSON_UTF_8).build();
         final List<FieldInfo> fieldInfos = ImmutableList.of(
-                new FieldInfo("header", "header", REQUIRED, toTypeSignature(int.class), "header parameter"),
-                new FieldInfo("query", "query", REQUIRED, toTypeSignature(long.class), "query parameter"));
+                new FieldInfo("header", "header", REQUIRED, INT32, "header parameter"),
+                new FieldInfo("query", "query", REQUIRED, INT64, "query parameter"));
         final MethodInfo methodInfo = new MethodInfo(
-                "foo", toTypeSignature(String.class), fieldInfos, ImmutableList.of(),
+                "foo", TypeSignature.ofBase("T"), fieldInfos, ImmutableList.of(),
                 ImmutableList.of(endpoint), HttpMethod.GET, "foo method");
         methodInfos.computeIfAbsent(MyService.class, unused -> new HashSet<>()).add(methodInfo);
     }
 
-    private static void addFooAllMethodInfos(Map<Class<?>, Set<MethodInfo>> methodInfos) {
-        final EndpointInfo endpoint = new EndpointInfoBuilder("*", "exact:/service/fooAll")
+    private static void addAllMethodsMethodInfos(Map<Class<?>, Set<MethodInfo>> methodInfos) {
+        final EndpointInfo endpoint = new EndpointInfoBuilder("*", "exact:/service/allMethods")
                 .availableMimeTypes(MediaType.JSON_UTF_8).build();
         Stream.of(HttpMethod.values())
               .filter(httpMethod -> httpMethod != HttpMethod.CONNECT && httpMethod != HttpMethod.UNKNOWN)
               .forEach(httpMethod -> {
                   final MethodInfo methodInfo =
-                          new MethodInfo("fooAll", toTypeSignature(String.class), ImmutableList.of(),
-                                         ImmutableList.of(), ImmutableList.of(endpoint), httpMethod, null);
+                          new MethodInfo("allMethods", TypeSignature.ofContainer("CompletableFuture",
+                                                                                 TypeSignature
+                                                                                         .ofUnresolved("")),
```

Comment:
Could break after `"allMethods"` to make more space for `TypeSignature.ofContainer()`.